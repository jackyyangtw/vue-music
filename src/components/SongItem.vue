<template>
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-white/[0.1]"
    :id="`song-${song.docID}`"
  >
    <div class="mr-3 max-w-[80%]">
      <router-link
        :to="{ name: 'song', params: { id: song.docID } }"
        class="font-bold block text-white composition-name"
      >
        {{ song.modifiedName }}
      </router-link>
      <span class="text-white text-sm composition-author">{{
        song.displayName
      }}</span>
    </div>

    <div class="text-white text-lg">
      <router-link
        custom
        :to="{ name: 'song', params: { id: song.docID }, hash: '#comments' }"
        v-slot="{ navigate }"
      >
        <span class="comments text-sm md:text-lg" @click="navigate">
          <i class="fa fa-comments text-white"></i>
          {{ song.commentCount }}
        </span>
      </router-link>
    </div>
  </li>
</template>

<script>
export default {
  name: "SongItem",
  props: ["song"],
};
</script>

<style lang="scss" scoped></style>
