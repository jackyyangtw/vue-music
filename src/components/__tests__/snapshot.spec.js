import { shallowMount } from "@vue/test-utils";
import SongItem from "../songItem.vue";
import { describe } from "vitest";
import { RouterLinkStub } from "@vue/test-utils";
describe("Snapshots SongItem.vue", () => {
  test("render correctly", () => {
    const song = {
      displayName: "test",
      docID: "abs",
      modifiedName: "test",
      commentCount: 5,
    };
    const wrapper = shallowMount(SongItem, {
      props: { song },
      global: {
        components: {
          "router-link": RouterLinkStub,
        },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
