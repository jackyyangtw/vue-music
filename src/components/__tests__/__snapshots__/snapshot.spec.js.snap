// Vitest Snapshot v1

exports[`Snapshots SongItem.vue > render correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-white/[0.1]"
  id="song-abs"
>
  <div
    class="mr-3 max-w-[80%]"
  >
    <router-link-stub
      class="font-bold block text-white composition-name"
      custom="false"
      to="[object Object]"
    />
    <span
      class="text-white text-sm composition-author"
    >
      test
    </span>
  </div>
  <div
    class="text-white text-lg"
  >
    <router-link-stub
      custom="true"
      to="[object Object]"
    />
  </div>
</li>
`;
